File: app/src/main/java/com/anetos/parkme/data/database/AppDatabase.java
package com.anetos.parkme.data.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.anetos.parkme.data.database.dao.NoteDao;
import com.anetos.parkme.data.model.Note;


@Database(
        entities = {Note.class},
        version = 2,
        exportSchema = false
)
public abstract class AppDatabase extends RoomDatabase {

    private static final String DB_NAME = "db_anetosnote";
    private static AppDatabase appDb = null;

    public static AppDatabase getInstance(Context context) {
        if (appDb == null) {
            appDb = buildDatabaseInstance(context);
        }
        return appDb;
    }

    private static AppDatabase buildDatabaseInstance(Context context) {
        return Room.databaseBuilder(
                context,
                AppDatabase.class,
                DB_NAME
        ).allowMainThreadQueries()
                .fallbackToDestructiveMigration()
                .build();
    }

    abstract NoteDao noteDao();
}

File: app/src/main/java/com/anetos/parkme/data/db/AppDatabase.kt
package  com.anetos.parkme.data.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase


//@Database(entities = [/*Location::class, SafeRoute::class*/], version = 6, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun locationDao(): LocationDao

    abstract fun routeDao(): RouteDao

    companion object {
        private const val DB_NAME = "Bangalore-Weather1"
        private var appDb: AppDatabase? = null

        fun getInstance(context: Context): AppDatabase {
            if (null == appDb) {
                appDb = buildDatabaseInstance(context)
            }
            return appDb as AppDatabase
        }

        private fun buildDatabaseInstance(context: Context): AppDatabase {
            return Room.databaseBuilder(
                context,
                AppDatabase::class.java,
                DB_NAME
            ).allowMainThreadQueries()
                .fallbackToDestructiveMigration()
                .build()
        }
    }
}